### PR TITLE
Update CI to cache `node_modules` per OS

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Does what the title says

## Testing

Testing on this PR

## Docs

N/A

### CODE REVIEW

**Comment by matthewp** [comment]
I was about to do the same thing :)

**Comment by matthewp** [comment]
Might just try rerunning the CI that's failing, looks like it could be a web service that's down.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,8 +64,8 @@ jobs:
           auto_fix: true
           git_name: github-actions[bot]
           git_email: github-actions[bot]@users.noreply.github.com
-          commit_message: 'chore(lint): ${linter} fix'
-          github_token: ${{ secrets.GITHUB_TOKEN }}
+          commit_message: '[ci] ${linter} fix'
+          github_token: ${{ secrets.NIGHTLY_PERSONAL_GITHUB_TOKEN }}
```
Comment (matthewp) [inline .github/workflows/ci.yml:68]: I'm assuming this token isn't set up yet. Is there a reason why we can't use the other?

Comment (natemoo-re) [reply .github/workflows/ci.yml:68]: Nope, we can revert!

**Review by jonathantneal — APPROVED** [review]

**Pull request merged**